docs/notebooks/feasible_sets.pct.py

```diff
@@ -0,0 +1,224 @@
+
+# %%
+from util.plotting import plot_bo_points, plot_function_2d
+import tensorflow_probability as tfp
+
+
+def excursion_probability(x, model, threshold=80):
+    mean, variance = model.model.predict_f(x)
+    normal = tfp.distributions.Normal(tf.cast(0, x.dtype), tf.cast(1, x.dtype))
+    t = (mean - threshold) / tf.sqrt(variance)
+    return normal.cdf(t)
+
+
+def plot_excursion_probability(title, model = None, query_points = None):
+    
+    if model is None:
+        objective_function = thresholded_branin
+    else:
+        def objective_function(x):
+            return excursion_probability(x, model)
+
+    _, ax = plot_function_2d(
+        objective_function,
+        search_space.lower - 0.01,
+        search_space.upper + 0.01,
+        grid_density=300,
+        contour=True,
+        colorbar=True,
+        figsize=(10, 6),
+        title=[title],
+        xlabel="$X_1$",
+        ylabel="$X_2$",
+    )
+    if query_points is not None:
+        plot_bo_points(query_points, ax[0, 0], num_initial_points)
+
+plot_excursion_probability("Excursion set, Branin function")
+
+
+# %% [markdown]
+# Next we illustrate the excursion probability map using the model fitted to the initial data alone. On the figure below we can see that the failure region boundary has been identified with some accuracy in the upper right corner, but not in the lower left corner. It is also loosely defined, as indicated by a slow decrease and increase away from the 0.5 excursion probability contour.
+
+# %%
+# extracting the data to illustrate the points
+dataset = result.try_get_final_dataset()
+query_points = dataset.query_points.numpy()
+observations = dataset.observations.numpy()
+
+# fitting the model only to the initial data
+initial_model = build_model(initial_data)
+initial_model.optimize(initial_data)
+
+plot_excursion_probability(
+    "Probability of excursion, initial data", initial_model, query_points[:num_initial_points,]
+)
+
+
+# %% [markdown]
+# Next we examine an updated excursion probability map after the 10 active learning steps. We can now see that the model is much more accurate and confident, as indicated by a good match with the reference thresholded Branin function and sharp decrease/increase away from the 0.5 excursion probability contour.
+
+# %%
+updated_model = result.history[-1].models["OBJECTIVE"]
+
+plot_excursion_probability(
+    "Updated probability of excursion", updated_model, query_points
+)
+
+
+# %% [markdown]
+# We can also examine what would happen if we would continue for many more active learning steps. One would expect that choices would be allocated closer and closer to the boundary, and uncertainty continuing to collapse. Indeed, on the figure below we observe exactly that. With 90 observations more the model is precisely representing the failure region boundary. It is somewhat difficult to see on the figure, but the most of the additional query points lie exactly on the threshold line. 
+
+# %%
+num_steps = 90
```

Comment:
wouldn't this high number slow down trieste build? maybe it's OK

Reply:
we now have a way of dealing with this, but its also fairly quick 1-2 mins